content/en/blog/2024/policing-a-sandbox.md

```diff
@@ -0,0 +1,296 @@
+        storage.source == allowed_image
+    }
+}
+
+ExecProcessRequest if {
+    input_command = concat(" ", input.process.Args)
+	some allowed_command in policy_data.allowed_commands
+	input_command == allowed_command
+}
+
+policy_data := {
+	"allowed_commands": [
+		"whoami",
+		"false",
+		"curl -s http://127.0.0.1:8006/aa/token?token_type=kbs",
+	],
+	"allowed_images": [
+		"pause",
+		"docker.io/library/nginx@sha256:e56797eab4a5300158cc015296229e13a390f82bfc88803f45b08912fd5e3348",
+	],
+}
+```
+
+Policy are in many cases dynamic and specific to the workload. Kata ships the [genpolicy tool](https://github.com/kata-containers/kata-containers/tree/365df81d5e98d4cb4f7c2e2e271b9fb503bc4e4c/src/tools/genpolicy) that will generate a reasonable default policy based on a given k8s manifest, which can be further refined by the user. A dynamic policy cannot be bundled in the rootfs, at least not fully, since it needs to be tailored to the workload. This implies we need to provide the Guest VM with the policy at launch time, in a way that allows us to trust the policy to be genuine and unaltered. In the next section we'll discuss how we can achieve this.
+
+## Init-Data
+
+Confidential Containers need to access configuration data that for practical reasons cannot be baked into the OS image. This data includes URIs and certificates required to access Attestation and Key Broker Services, as well as the policy that is supposed to be enforced by the policy engine. This data is not secret, but maintaining its integrity is crucial for the confidentiality of the workload.
+
+In the CoCo project This data is referred to as [Init-Data](https://github.com/confidential-containers/trustee/blob/162c620fd9bcd8d6db4bb5b0a5944932a160e89f/kbs/docs/initdata.md). Init-data is specified as file/content dictionary in the TOML language, optimized for easy authoring and human readability. Below is a (shortened) example of a typical Init-Data block, containing some pieces of metadata, configuration for CoCo guest components and a policy in Rego language:
+
+```toml
+algorithm = "sha256"
+version = "0.1.0"
+
+[data]
+"aa.toml" = '''
+[token_configs]
+[token_configs.kbs]
+url = 'http://my-as:8080'
+cert = """
+-----BEGIN CERTIFICATE-----
+MIIDEjCCAfqgAwIBAgIUZYcKIJD3QB/LG0FnacDyR1KhoikwDQYJKoZIhvcNAQEL
+...
+4La0LJGguzEN7y9P59TS4b3E9xFyTg==
+-----END CERTIFICATE-----
+"""
+'''
+
+"cdh.toml"  = '''
+socket = 'unix:///run/confidential-containers/cdh.sock'
+credentials = []
+...
+'''
+
+"policy.rego" = '''
+package agent_policy
+...
+'''
+
+```
+
+A user is supposed to specify Init-Data to a Confidential Guest in the form of a base64-encoded string in a specific Pod annotation. The runtime will then pass this data on to the Agent in the Guest VM, which will decode the Init-Data and use it to configure the runtime environment of the workload. Crucially, since Init-Data is not trusted at launch we need a way to establish that the policy has not been tampered with in the process.
```

Comment:
The runtime will then pass ...

There are way too many uses or runtime, which may confuse the reader a little bit.  I think you could use "the container runtime (iow, kata-containers)", or simply "kata containers will then pass" ...